src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs

```diff
@@ -701,8 +709,45 @@ private bool ReplacementBreaksQueryClause(QueryClauseSyntax originalClause, Quer
                 !SymbolInfosAreCompatible(originalClauseInfo.OperationInfo, newClauseInfo.OperationInfo);
         }
 
+        private static bool IsPotentiallyTargetTypedConditionalExpression(ExpressionSyntax expressionSyntax)
+        {
+            return expressionSyntax is ConditionalExpressionSyntax &&
+                   ((CSharpParseOptions)expressionSyntax.SyntaxTree.Options).LanguageVersion >= LanguageVersion.CSharp9;
+        }
+
+        protected override bool ReplacementIntroducesErrorType(ExpressionSyntax originalExpression, ExpressionSyntax newExpression)
+        {
+            // The base implementation will see that the type of the new expression may potentially change to null,
+            // because the expression has no type but can be converted to a conditional expression type. In that case,
+            // we don't want to consider the null type to be an error type.
+            if (IsPotentiallyTargetTypedConditionalExpression(newExpression) &&
+                this.SpeculativeSemanticModel.GetConversion(newExpression).IsConditionalExpression)
+            {
+                return false;
+            }
+            else
+            {
+                return base.ReplacementIntroducesErrorType(originalExpression, newExpression);
+            }
+        }
+
         protected override bool ConversionsAreCompatible(SemanticModel originalModel, ExpressionSyntax originalExpression, SemanticModel newModel, ExpressionSyntax newExpression)
-            => ConversionsAreCompatible(originalModel.GetConversion(originalExpression), newModel.GetConversion(newExpression));
+        {
+            var originalConversion = originalModel.GetConversion(originalExpression);
+            var newConversion = newModel.GetConversion(newExpression);
+
+            if (IsPotentiallyTargetTypedConditionalExpression(originalExpression) &&
+                IsPotentiallyTargetTypedConditionalExpression(newExpression) &&
+                ((originalConversion.IsIdentity && newConversion.IsConditionalExpression) ||
+                 (originalConversion.IsConditionalExpression && newConversion.IsIdentity)))
```

Comment:
do we have tests for both of tehse directions?

Reply:
I added 742 more out of paranoia since it felt like only having it in one direction seemed odd. I think I have some ideas for tests to try though.

Reply:
More ideas for tests failed, so I removed the conditional to identity case just to be conservative with the change.